#! /bin/sh -e

# DP: Set HeadURL and PY_PATCHLEVEL_REVISION.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

--- Python/sysmodule.c~	2006-10-29 11:48:50.000000000 +0100
+++ Python/sysmodule.c	2006-10-29 12:26:53.000000000 +0100
@@ -960,7 +960,7 @@
 
 /* Subversion branch and revision management */
 static const char _patchlevel_revision[] = PY_PATCHLEVEL_REVISION;
-static const char headurl[] = "$HeadURL: svn+ssh://pythondev@svn.python.org/python/tags/r25/Python/sysmodule.c $";
+static const char headurl[] = "$HeadURL: svn+ssh://pythondev@svn.python.org/python/branches/release25-maint/Python/sysmodule.c $";
 static int svn_initialized;
 static char patchlevel_revision[50]; /* Just the number */
 static char branch[50];
--- Include/patchlevel.h~	2006-09-18 08:51:50.000000000 +0200
+++ Include/patchlevel.h	2006-10-29 12:33:07.000000000 +0100
@@ -29,7 +29,7 @@
 #define PY_VERSION		"2.5"
 
 /* Subversion Revision number of this file (not of the repository) */
-#define PY_PATCHLEVEL_REVISION  "$Revision: 54692 $"
+#define PY_PATCHLEVEL_REVISION  "$Revision: 54692 $"
 
 /* Version as a single 4-byte hex number, e.g. 0x010502B2 == 1.5.2b2.
    Use this for numeric comparisons, e.g. #if PY_VERSION_HEX >= ... */
